### ISSUE TITLE
LTI Mass Update not work with error Missing required parameter 'consumer_secret'

### ISSUE THREAD

**Issue by somiaj** [openwebwork/webwork2#1989]
I'm using the 2.18 RC for a tiny class to test things out, and I noticed that the LTI mass update is no longer working and the `webwork2.log` file is giving me:

```
[error] An error occured while trying to mass update grades via LTI: Missing required parameter 'consumer_secret' at /opt/webwork/webwork2.18/lib/WeBWorK/Authen/LTIAdvanced/SubmitGrade.pm line 408.
```

But LTI auth and LTI grade pass back on problem submit is working just fine for this course. My initial guess by looking at the error and seeing that LTI grade passback is working on each submit is that the minion queue might have issue getting the full course environment? Since this is a live class unsure how to nicely debug the issue.

**Comment by drgrice1** [comment]
The minion job queue should have no problem obtaining the full course environment for the course.  What is really odd is the line number of our error.  There is no warning like that anywhere to be found in the code, not to mention anywhere near line 408 of `lib/WeBWorK/Authen/LTIAdvanced/SubmitGrade.pm`.

**Comment by somiaj** [comment]
Agreed, I didn't find anything useful near that line number looking at the code on the server either. I did double check that `consumer_secret` was being defined in the hashes above, which it is from the course environment, so that is why I was wondering if there is some issue there. Grades are passing back, just not via LTI mass update.

Once my students aren't being that active I can poke around the server some more if that could help debug this. I'll start by turning on LTI debugging and see if I get anything useful in the logs. Any other suggestions of where to debug this?

**Comment by drgrice1** [comment]
I just did a little testing again, but the task in the job queue is certainly getting the consumer secret from the settings.  Logging `$ce->{LTI}{v1p1}{BasicConsumerSecret}` from the `submit_grade` in SubmitGrade.pm shows this.

**Comment by somiaj** [comment]
Would setting `$LTI{v1p1}{BasicConsumerSecret}` in `simple.conf` have any affect on this? I'll debug later tonight once I'm home and see what other info I can find.

**Comment by drgrice1** [comment]
It would work to set it from `simple.conf`.   I tested this and grade passback in general, and it all seems to be working for me.

**Comment by drgrice1** [comment]
I thought of something that could be happening.  You have had webwork 2.18 running for a while on this server right?  Since before we merged the LTI 1.3 stuff?  Have you pulled in those changes?  If so, after doing that you probably restarted the webwork2 service with `sudo systemctl restart webwork2`, but did you also restart the job queue service with `sudo systemctl restart webwork2-job-queue`?  If you didn't do that, then the job queue would still be using the code from before the update.

**Comment by somiaj** [comment]
@drgrice1 Thanks, that was it. I forgot to turn it off, then turn it back on again.
